py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,512 @@
+Examples:
+```
+# add color gradients for events and dates/times
+google_calendar {
+    thresholds = {
+        'event': [(1, '#d0e6ff'), (2, '#bbdaff'), (3, '#99c7ff'), (4, '#86bcff'),
+            (5, '#62a9ff'), (6, '#8c8cff'), (7, '#7979ff')],
+        'time': [(1, '#ffcece'), (2, '#ffbfbf'), (3, '#ff9f9f'), (4, '#ff7f7f'),
+            (5, '#ff5f5f'), (6, '#ff3f3f'), (7, '#ff1f1f')]
+    }
+}
+```
+
+@author Igor Grebenkov
+@license BSD
+
+SAMPLE OUTPUT
+[
+    {
+      'full_text':'Homer's Birthday (742 Evergreen Terrace) (1h 23m)',
+      'index':0,
+      'separator':False,
+      'separator_block_width':0,
+      'cached_until':1509008337.0
+   },
+   {
+      'full_text':'|',
+      'index':'sep',
+      'separator':False,
+      'separator_block_width':0
+   },
+   {
+      'full_text':'Doctor's Appointment',
+      'index':1,
+      'separator':False,
+      'separator_block_width':0
+   },
+   {
+      'full_text':'|',
+      'index':'sep',
+      'separator':False,
+      'separator_block_width':0
+   },
+   {
+      'full_text':'Lunch with John',
+      'index':2
+   }
+]
+"""
+
+import httplib2
+import os
+import datetime
+import pytz
+
+from apiclient import discovery
+from oauth2client import client
+from oauth2client import clientsecrets
+from oauth2client import tools
+from oauth2client.file import Storage
+from httplib2 import ServerNotFoundError
+
+
+SCOPES = 'https://www.googleapis.com/auth/calendar.readonly'
+APPLICATION_NAME = 'py3status google_calendar module'
+
+
+class Py3status:
+    auth_token = '~/.config/py3status/google_calendar.auth_token'
+    button_open = 3
+    button_refresh = 2
+    button_toggle = 1
+    cache_timeout = 60
+    client_secret = '~/.config/py3status/google_calendar.client_secret'
+    force_lowercase = False
+    format = '{events}| \?color=event \u2687'
+    format_date = '%a %d-%m'
+    format_event = '\?color=event {summary}[\?if=location  ({location})][\?if=time_to  {time_to}]'
+    format_event_full = '[\?color=event {summary}] ' +\
+        '[\?color=time ({start_time} - {end_time}, {start_date})]'
+    format_separator = '\?color=separator \|'
+    format_time = '%I:%M %p'
+    format_time_left = '\?color=time ([\?if=days {days}d ]' +\
+        '[\?if=hours {hours}h ][\?if=minutes {minutes}m]) left'
+    format_time_to = '\?color=time ([\?if=days {days}d ]' +\
+        '[\?if=hours {hours}h ][\?if=minutes {minutes}m])'
+    ignore_all_day_events = False
+    num_events = 3
+    thresholds = []
+    time_to_max = 180
+    warn_threshold = 0
+    warn_timeout = 300
+
+    def post_config_hook(self):
+        self.button = None
+        self.button_states = [False] * self.num_events
+        self.events = None
+        self.no_update = False
+
+        self.client_secret = os.path.expanduser(self.client_secret)
+        self.auth_token = os.path.expanduser(self.auth_token)
+
+        self.credentials = self._get_credentials()
+        self.is_authorized = self._authorize_credentials()
```

Comment:
(Opinion) I think this should be `False` so you can use the one in the loop instead.

Reply:
Can you clarify please?

Reply:
See how `google_calendar` operates offline. If `google_calendar` is still trying, then modules won't show up on the bar until everything is finished... is which we may want to do this... or even to return nothing in the first loop.
* `github` return `?`
* `coin_market`  return the format. 
* `i3block` return `''`

Make `self.is_authorized = False` in `post_config_hook` to not do the run in `post_config_hook` because `authorize_credentials` could be hanging in the `post_config_hook` offline... and it is to reutrn the `cute icon` first thing first (just like gcal) as it can obtain the data in next loop.

Reply:
EDIT: We don't have to do this if you think it responds correctly to offline or w/e.